csharp/src/Google.Protobuf/WellKnownTypes/DurationPartial.cs

```diff
@@ -266,5 +266,26 @@ internal static void AppendNanoseconds(StringBuilder builder, int nanos)
                 }
             }
         }
+
+
+        /// <summary>
+        /// Given another duration, returns 0 if the durations are equivalent, -1 if this duration is shorter than the other, and 1 otherwise.
+        /// </summary>
+        /// <remarks>
+        /// This method expects that both durations are normalized; that is, that the values of <see cref="Seconds"/>
+        /// and <see cref="Nanos"/> are within the documented bounds.
+        /// If either value is not normalized, the results of this method are unspecified.
+        /// </remarks>
+        /// <param name="other">The duration to compare with this object.</param>
+        /// <returns>An integer indicating whether this duration is shorter or longer than <paramref name="other"/>.</returns>
+        public int CompareTo(Duration other)
```

Comment:
Unknown fields are not being taken into account here but they are being taken into account for equality and hashcode. Not sure how to take them into account but it may happen that something equals here is then not equals with `Equals`.

Maybe a note in the docs is enough.

Reply:
Given that this is a well-known type, we really shouldn't be getting any unknown fields - I think it's fine for the comparison to ignore them.